howto/stream/exchange-oob-data.md

```diff
@@ -24,83 +24,111 @@ The following instructions guide you to exchange OOB data using a specific imple
 ```
-let stream = new AnboxStream({
-  ...
-  ...
-  dataChannels: {
-    "foo": {
-      callbacks: {
-        close: () => {
-          console.log('data channel is closed')
-        },
-        open: () => {
-          console.log('data channel is open')
-        },
-        error: (err) => {
-          console.log(`error: ${err}`)
-        },
-        message: (data) => {
-          console.log(`data received: ${data}`)
-        }
-      }
-    }
-  }
-});
+  <body>
+    <div id="anbox-stream"></div>
+    <div id="controls" style="position: absolute; top: 20px; left: 20px; z-index: 10">
+      <input type="text" id="textBox" placeholder="Send message">
+      <button id="sendButton">Send</button>
+    </div>
+
+    <script>
+      ...
+      window.onload = () => {
+        const stream = new AnboxStream({
+          targetElement: "anbox-stream",
+          ...
+          dataChannels: {
+            "foo": {
+              callbacks: {
+                close: () => {
+                  console.log('data channel is closed')
+                },
+                open: () => {
+                  console.log('data channel is open')
+                },
+                error: (err) => {
+                  console.log(`error: ${err}`)
+                },
+                message: (data) => {
+                  console.log(`data received: ${data}`)
+                }
+              }
+            }
+          }
+        });
+        stream.connect();
+
+        const sendButton = document.getElementById("sendButton");
+        const textBox = document.getElementById("textBox");
+        sendButton.addEventListener("click", () => {
+          const message = textBox.value;
+          if (message) {
+            stream.sendData('foo', message);
+          }
+        });
+      };
+    </script>
+  </body>
+
 ```
 
+1. This extends a `AnboxStream` object to create a data channel named `foo` and registers event handlers for data communication between the Anbox instance and web client.
+2. It also adds a new section to the web page, containing an input field to type and send a message to Anbox runtime through the created data channel.
+
+
+
 ```{note}
 An `AnboxStream` object can create a maximum of five data channels. If the number of data channels exceeds the allowed maximum, an exception is thrown when instantiating the `AnboxStream` object.
 ```
 
-To launch a new WebRTC session, the client must call `stream.connect()`. The `AnboxStream` object then builds up a WebRTC native data channel internally, based on the name declared under its `dataChannels` property for peer-to-peer communication (`foo` in the example).
+### Data exchange between Anbox runtime and web client
 
-To send data to an Android application through the channel, the client must use the member function `sendData` of the `AnboxStream` class:
+Launch a stream-enabled instance
 
 ```
-stream.sendData('foo', 'hello world');
+amc launch --name test-app \
+  --raw jammy:android12:amd64 \
```

Comment:
We should use a newer image and hint to the user that they may need to select `arm64` instead.